awscrt/auth.py

```diff
@@ -255,6 +255,32 @@ def new_chain(cls, providers):
         binding = _awscrt.credentials_provider_new_chain(providers)
         return cls(binding)
 
+    @classmethod
+    def new_python(cls, py_provider):
+        """
+        Creates a provider that sources credentials from a python class.
```

Comment:
Not necessary at this time, but in the future we should support async delegate get_credentials functions.
Like, if the user return an `AwsCredentials` then we'll do what we're doing here, but if they return a `Future` then we add a done_callback to the future that passes the ultimate result down to C.